pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -377,6 +377,49 @@ func (c *Cloud) ensureLoadBalancerHealthCheck(loadBalancer *elb.LoadBalancerDesc
 	return nil
 }
 
+// Makes sure that the health check for an ELB matches the configured health check node port
+func (c *Cloud) ensureHttpHealthCheck(loadBalancer *elb.LoadBalancerDescription, path string, port int32) error {
+	name := aws.StringValue(loadBalancer.LoadBalancerName)
+
+	actual := loadBalancer.HealthCheck
+
+	// Default AWS settings
+	expectedHealthyThreshold := int64(2)
+	expectedUnhealthyThreshold := int64(6)
+	expectedTimeout := int64(5)
+	expectedInterval := int64(10)
+
+	expectedTarget := "HTTP:" + strconv.FormatInt(int64(port), 10) + path
+
+	if expectedTarget == orEmpty(actual.Target) &&
```

Comment:
AFAICT from this point on the two health-check configuration functions are the same (there's a lot of overlap earlier also!)  Can we combine the two tails here into a single function?  Maybe even just have one function, and build the expected healthCheck in EnsureLoadBalancer ?